Scenarios/SetupPowerPlatform.md

```diff
@@ -0,0 +1,63 @@
+# Connect your GitHub repository to Power Platform
+
+There are 2 steps to connect your GitHub repository to your Power Platform tenant and environments:
+
+1. Setup the authentication context
+2. Setup the AL-Go Repository settings (.github/AL-Go-Settings.json)
+
+## Authentication Context
+The authentication context is a JSON object that you save in your GitHub secrets with the following naming convention: `<GitHubEnvironmentName>_AUTHCONTEXT`
+
+The authentication context specifies how the GitHub Environment you have created authenticates against your Power Platform and Business Central environments. You can use UserName/Password to create the authentication context or create a service principal account and use a client secret to connect (see a [Setting up service principal](./SetupServicePrincipalForPowerPlatform.md) for detailed steps).
+
+> **NOTE:** Username/password authentication is only supported for tenants that do NOT have 2-factor authentication enabled.
+The recommended way to get the auth context is to use the BCContainerHelper to generate the JSON - open a PowerShell window and run the following commands:
+
+### Getting service principal authentication context
+
+```powershell
+# If not already installed, install latest BcContainerHelper
+Install-Module BcContainerHelper -force
+$ppClientId = Read-Host -Prompt "Enter client id"
+$ppClientSecret = Read-Host -AsSecureString -Prompt 'Enter client secret'
+New-BcAuthContext -includeDeviceLogin | New-ALGoAuthContext -ppClientSecret $ppClientSecret -ppApplicationId $ppClientId | Set-Clipboard
+```
+
+### Getting username/password authentication context:

```

Comment:
Yes

Reply:
What is the need? 😄 The more of these scenarios we support the more we also have to maintain. 

If both types of auth allow the developers to do the same thing, why not just stick with one of them?

Reply:
ClientID/ClientSecret is the preferred mechanism for Production etc. - we cannot live without that. Username/Password is preferred for dev/test, it would be hard to live without that.